### PR TITLE
ci: update release script & add nightly releases

### PR DESCRIPTION
**Pull request by szymonrybczak**
Summary:
---------

Creates `release-nightly.yml` script which check if there are any new commits presented in last 24 hours on branch if yes, it publishes new prerelease. 

Test Plan:
----------

Checklist
----------

- [x] Documentation is up to date to reflect these changes.
- [x] Follows commit message convention described in [CONTRIBUTING.md](https://github.com/react-native-community/cli/blob/main/CONTRIBUTING.md#commit-message-convention)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,48 +1,42 @@
 name: Release
 
-# Temporarily disable until issues resolved
 on:
-#   release:
-#     types: [published]
+  release:
+    types: [published]
```
Comment (szymonrybczak) [inline .github/workflows/release.yml:5]: Q: Should it stay like that, or should we create other convention for creating releases?

File: .github/workflows/nigthly-release.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20'
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: 'Setup git user'
+        run: |
+          git config user.name "${{ github.actor }}"
+          git config user.email "bot@callstack.io"
```
Comment (szymonrybczak) [inline .github/workflows/nigthly-release.yml:30]: @cortinico let me know under what e-mail address `react-native-community-bot` will be :pray:

Comment (cortinico) [reply .github/workflows/nigthly-release.yml:30]: We've been using `bot@reactnative.dev` for commit's sake, but we don't have access to that email FYI

File: .github/workflows/nigthly-release.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
```
Comment (szymonrybczak) [inline .github/workflows/nigthly-release.yml:30]:
```suggestion
          git config user.email "bot@reactnative.dev"
```

File: .github/workflows/nigthly-release.yml (hunk 2 above)
Comment (szymonrybczak) [reply .github/workflows/nigthly-release.yml:30]: alright, let's use it 👍

**Review by cortinico — APPROVED** [review]

File: .github/workflows/nigthly-release.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+          else
+             echo "::set-output name=changed::false"
+          fi
+
+      - name: 'Prelease'
+        if: steps.check.outputs.changed == 'true'
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          yarn run publish:next
```
[the remark this answers is not in the document]
Comment (szymonrybczak) [reply .github/workflows/nigthly-release.yml]: Finally followed the same strategy as `react-native` repo

**Review by thymikee — APPROVED** [review]

File: .github/workflows/nigthly-release.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+          # If PATCH_VERSION contains a '-', strip everything after it
+          if echo $PATCH_VERSION | grep -q '-'; then
+              PATCH_VERSION=${PATCH_VERSION%%-*}
+          fi
+
+          # Create a version string that includes the commit hash and date
+          VERSION="${MAJOR_VERSION}.${MINOR_VERSION}.${PATCH_VERSION}-${PRERELEASE_NAME}-${DATE}-${COMMIT_HASH}"
+
+          npx lerna version --force-publish --no-push --allow-branch '*' --preid $VERSION --yes
+          npx lerna publish --tag nightly from-git --no-verify-access --no-changelog --yes
```
Comment (thymikee) [inline .github/workflows/nigthly-release.yml:69]: I’d maybe move the “from-git” argument before flags for readability but that’s not too important

Comment (szymonrybczak) [reply .github/workflows/nigthly-release.yml]: https://github.com/react-native-community/cli/pull/2440/commits/37b33ccec12781d1f981a7fab439e3d7ed757cdb

**Pull request merged**